Modules/TPC/src/IDCs.cxx

```diff
@@ -163,18 +172,18 @@ void IDCs::update(Trigger, framework::ServiceRegistryRef)
   mCCDBHelper.setFourierCoeffs(idcFFTA, Side::A);
   mCCDBHelper.setFourierCoeffs(idcFFTC, Side::C);
 
-  if (idcZeroA) {
+  if (idcZeroA && idcZeroC) {
+    // scale IDCZero to the sum of IDCZeros
+    mCCDBHelper.setIDCZeroScale(true);
+    mCCDBHelper.drawIDCZeroScale(mIDCZeroScale.get(), true);
     mCCDBHelper.drawIDCZeroStackCanvas(mIDCZeroStacksA.get(), Side::A, "IDC0", mRanges["IDCZero"].at(0), mRanges["IDCZero"].at(1), mRanges["IDCZero"].at(2));
-  }
-
-  if (idcZeroC) {
     mCCDBHelper.drawIDCZeroStackCanvas(mIDCZeroStacksC.get(), Side::C, "IDC0", mRanges["IDCZero"].at(0), mRanges["IDCZero"].at(1), mRanges["IDCZero"].at(2));
-  }
-
-  if (idcZeroA && idcZeroC) {
     mCCDBHelper.drawIDCZeroRadialProfile(mIDCZeroRadialProf.get(), mRanges["IDCZero"].at(0), mRanges["IDCZero"].at(1), mRanges["IDCZero"].at(2));
   }
-
+  const auto& calDet = mCCDBHelper.getIDCZeroCalDet();
+  if (&calDet) {
+    o2::tpc::painter::draw(calDet, mRanges["IDCZeroOveview"].at(0), mRanges["IDCZeroOveview"].at(1), mRanges["IDCZeroOveview"].at(2), mIDCZerOverview.get());
+  }
```

Comment:
The function returns an object. References cannot be nullpointer
```suggestion
  const auto calDet = mCCDBHelper.getIDCZeroCalDet();
  o2::tpc::painter::draw(calDet, mRanges["IDCZeroOveview"].at(0), mRanges["IDCZeroOveview"].at(1), mRanges["IDCZeroOveview"].at(2), mIDCZerOverview.get());
```

Reply:
changed! thanks a lot